File: Minimum Steps Required.cpp
int minSteps(string s) {
    int i=0;
    int ca=0;
    int cb=0;
    for(int i=0;i<s.size();i++){
        if(i==0){
            if(s[i]=='a')ca=1;
            else cb=1;
            continue;
        }
        if(s[i]==s[i-1])continue;
        if(s[i]=='a')ca++;
          else cb++;
    }
    int ans=0;
     return min(ca,cb)+1;
    
}

File: Minimum Steps Required.java
class Solution{
	int minSteps(String str) {
		//Write your code here
	// Get the length of the input string
    int n = str.length();
    // Initialize a counter variable to keep track of the number of changes
    int count = 0;
    // Initialize a variable to keep track of the previous character encountered
    char prev = '\0';
    // Loop through the string character by character
    for(int i=0; i<n; i++) {
        // Get the current character
        char ch = str.charAt(i);
        /*If the current character is different from the previous character, 
        increment the counter and update the previous character*/
        if(ch != prev) {
            count++;
            prev = ch;
        }
    }
    return count/2+1;
	}
}
